### DISCUSSION TITLE
NextJS App Router Microfrontend approach

### DISCUSSION THREAD

**Discussion by develos-dr** [vercel/next.js#66563] [Help]
### Summary

Hi!

Can you tell me what is a proper approach to implement Microfrontend in Next JS 14+ with App Router approach? I already read a lot about module federation plugin for next but its only for pages, not for App Router. I assume that the app router is and will be main apprach and soon pages will be deprecated so i dont see a reason to write new application with pages when app router is a base approach. But we want to integrate different projects with MFA approach and i dont know what is the one for app router. 

Thank you in advanced for your time and answers!

Best Regards!

**Comment by ltkropidlowski** [comment]
I am interested in this topic too!

**Comment by faaizSayed03** [comment]
I'm also looking for the same, @develos-dr Did you found anything?

**Comment by mdadil09** [comment]
I am interested in this, @develos-dr Did you found anything?

**Comment by TheKachi** [comment]
I am also interested in this but it seems like there's currently nothing that can be done which is sad

**Comment by apipkurniawan** [comment]
I am interested in this, @develos-dr  Did you found anything?

**Comment by develos-dr** [comment]
No, I changed approach


Best Regards,



[Image]

Dawid Ratajczak

CEO

Mobile: +48 572-513-238
NIP: 604-022-87-29
REGON: 52121310300000
http://www.develos.pl/



Develos Ratajczak Gajos Spółka Komandytowo-Akcyjna, ul. Biała 4, 80-180 Jankowo Gdańskie
zarejestrowana w Sądzie Rejonowym Gdańsk-Północ w Gdańsku, VII Wydział Gospodarczy nr KRS: 0000952438.
Regon: 52121310300000 | Kapitał zakładowy: 50.000,00 zł | NIP: 604-022-87-29 | Nr konta bankowego: PLN 73 1140 1065 0000 5708 3500 1001 | USD: PL19 1140 1065 0000 5708 3500 1003 | EUR: PL46 1140 1065 0000 5708 3500 1002

Uwaga: Ten e-mail (wraz z załącznikami) jest poufny i przeznaczony tylko dla adresata(ów) tej wiadomości. Jeżeli nie jesteś adresatem niniejszej wiadomości, jej rozpowszechnianie, kopiowanie oraz rozprowadzanie(dystrybucja) jest zabronione. Jeśli wiadomość ta znalazła się w Twojej skrzynce w wyniku błędu, proszę poinformuj mnie o tym niezwłocznie i usuń oryginał wiadomości wraz z wszelkimi wydrukami(kopiami).

Important Note: This e-mail (including any attachments) is confidential and intended only for the use of the addressee (s) named herein. If you have received this e-mail in error, you are hereby notified that any review, copying or distribution of it is strictly prohibited. Please inform me immediately and destroy the original e-mail and any printouts.

________________________________
Od: apip kurniawan ***@***.***>
Wysłane: Wednesday, October 9, 2024 4:54:57 AM
Do: vercel/next.js ***@***.***>
DW: Dawid Ratajczak ***@***.***>; Mention ***@***.***>
Temat: Re: [vercel/next.js] NextJS App Router Microfrontend approach (Discussion #66563)


I am interested in this, @develos-dr<https://github.com/develos-dr> Did you found anything?

**Comment by Jero786** [comment]
Does anyone have any news about this?

**Reply by zackarychapple** [reply]
There is work in progress

**Reply by alzkdpf** [reply]
@zackarychapple Is there any ongoing news?

**Reply by zackarychapple** [reply]
I expect some significant movement before end of the month there is already a . PRs are starting to land for layers support.  There is an issue @ScriptedAlchemy opened in the MF core for testing https://github.com/module-federation/core/pull/3596
